_sc2data/Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/ActorData.xml

```diff
@@ -6846,4 +6903,212 @@
     <CActorModel id="AP_ColonyInfestationBuildingDeath" parent="UnitDeathModel">
         <On Terms="ActorCreation" Send="SetTintColor 200,160,133"/>
     </CActorModel>
+    <CActorSound id="AP_SOAHeroicShieldDeathSound" parent="SoundOneShot">
+        <On Terms="Behavior.AP_SOAHeroicShieldImmunity.Off" Send="Create"/>
+        <Sound value="AP_SOA_GuardianShell_End"/>
+    </CActorSound>
+    <CActorModel id="AP_SOAHeroicShieldLongModel" parent="ModelAnimationStyleContinuous">
+        <Inherits index="BaseModelScale" value="1"/>
+        <Inherits index="Scale" value="1"/>
+        <On Terms="Behavior.AP_SOAHeroicShieldImmunity.On; ValidateUnit IsCarrier" Send="Create"/>
+        <On Terms="Behavior.AP_SOAHeroicShieldImmunity.On; ValidateUnit IsVoidRay" Send="Create"/>
+        <On Terms="Behavior.AP_SOAHeroicShieldImmunity.On; ValidateUnit IsBattlecruiser" Send="Create"/>
+        <On Terms="Behavior.AP_SOAHeroicShieldImmunity.On; ValidateUnit AP_IsHercules" Send="Create"/>

```

Comment:
Check these.
in `AP_SOAHeroicShieldModel` you have
```xml
        <On Terms="ActorCreation; ValidateUnit AP_IsColossus" Send="Destroy"/>
        <On Terms="ActorCreation; ValidateUnit AP_IsCarrier" Send="Destroy"/>
        <On Terms="ActorCreation; ValidateUnit AP_IsVoidRay" Send="Destroy"/>
        <On Terms="ActorCreation; ValidateUnit IsBattlecruiser" Send="Destroy"/>
        <On Terms="ActorCreation; ValidateUnit AP_IsHercules" Send="Destroy"/>
```
The Colossus belongs to the Tall model. But the others matter